### DISCUSSION TITLE
`can` Middleware throwing 403 on my custom guard

### DISCUSSION THREAD

**Discussion by mePrazen** [spatie/laravel-permission#2451] [Q&A]
Route::get('/index', [IndexRestaurantUserController::class, 'index'])->name('index')->middleware('can:view_all_restaurant_user');
why is this middleware throwing 403 | This action is unauthorized while the user has the permission named view_all_restaurant_user.
And i am using this middleware for my custom guard.
But the same middleware works for default User model

**Comment by parallels999** [comment]
refresh spatie permissions cache
